### PR TITLE
Don't save scripts when exiting editor

### LINKED ISSUE #18010: Modified GDScript code gets saved to disk without a prompt
**Godot version:**
`master` branch commit eaa5dfbdbd1217d62902f966496658770638c3d1

**Steps to reproduce:**
1. Save all the changes to a scene.
2. Open a GDScript with the built-in editor and write some text to it and do not save those changes.
3. Go to Scene->Quit to exit the editor, and click on Quit.
4. The GDScript that was being modified gets saved to disk

I've also noticed that when editing a GDScript that was just saved, it is not getting marked as modified in the editor as soon as you start typing.

**Minimal reproduction project:**
[saved_without_asking.zip](https://github.com/godotengine/godot/files/1881884/saved_without_asking.zip)

### PR DESCRIPTION
**Pull request by KoBeWi**
Fixes #18010

The script editor properly tracks saved/unsaved scripts, so only unsaved scripts will get discarded.
Unfortunately the user won't be warned, see #55026

### CODE REVIEW

**Comment by akien-mga** [comment]
> Unfortunately the user won't be warned, see #55026

Should we aim to solve this before merging this PR?

I feel like saving too eagerly is better than losing changes unknowingly.

**Comment by KoBeWi** [comment]
I have a PR that resolves this issue: #67503, but it depends on yet another PR 🙃
But this can wait I guess.

**Review by akien-mga — APPROVED** [review]
LGTM, to be merged after #67503.

**Comment by YuriSizov** [comment]
Thanks!

**Pull request merged**
